proxy/proxy.go

```diff
@@ -408,10 +409,11 @@ func (dc *skipperDialer) DialContext(ctx stdlibcontext.Context, network, addr st
 			dialingFailed: true, // indicate error happened before http
 		}
 	} else if cerr := ctx.Err(); cerr != nil {
-		// deadline exceeded or canceled in stdlib
+		// deadline exceeded or canceled in stdlib, client closed request
+		// see https://github.com/zalando/skipper/issues/687#issuecomment-405557503
 		return nil, &proxyError{
-			err:  cerr,
-			code: http.StatusGatewayTimeout,
+			err:  fmt.Errorf("err from dial contrext: %v", cerr),
+			code: 499,
```

Comment:
Are we sure this is always caused by client closing the connection?

I'm just afraid that we are masking some `http: server closed idle connection` errors as mentioned in https://github.com/zalando/skipper/issues/687 but I'm also not sure where they came from.